src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -731,10 +756,20 @@ public void RegisterMessageHandler(Func<Message, CancellationToken, Task> handle
         protected virtual async Task<IList<Message>> OnReceiveAsync(int maxMessageCount, TimeSpan serverWaitTime)
         {
             ReceivingAmqpLink receiveLink = null;
+
+            if (this.isSessionReceiver)
```

Comment:
I see this block is repeated multiple times.
Would it help if `ThrowIfSessionLockLost()` would be invoked and condition would be within the method?